### PR TITLE
add trust domain to meshconfig

### PR DESCRIPTION
**Pull request by quanjielin**
https://github.com/istio/istio/issues/9035

later pilot needs to read meshconfig.trustdomain to construct secure naming

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -157,8 +157,12 @@ message MeshConfig {
   // will be used to generate key/cert eventually. This isn't supported for non-k8s case.
   bool enable_sds_token_mount = 23;
 
+  // The trust domain corresponds to the trust root of a system.
+  // Refer to https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain
```
Comment (diemtvu) [inline mesh/v1alpha1/config.proto]: What's the expected behavior if this is not set?

Comment (quanjielin) [reply mesh/v1alpha1/config.proto]: add comment ``` // Fallback to old identity format(without trust domain) if not set.```

**Pull request merged**